src/Core/Configurations/RuntimeConfigProvider.cs

```diff
@@ -103,12 +103,18 @@ private bool TrySetupConfigFileWatcher()
         {
             if (!IsLateConfigured && _runtimeConfig is not null && _runtimeConfig.IsDevelopmentMode())
             {
-                FileSystemRuntimeConfigLoader loader = (FileSystemRuntimeConfigLoader)ConfigLoader;
-                _configFileWatcher = new(this, loader.GetConfigDirectoryName(), loader.GetConfigFileName());
+                try
+                {
+                    FileSystemRuntimeConfigLoader loader = (FileSystemRuntimeConfigLoader)ConfigLoader;
+                    _configFileWatcher = new(this, loader.GetConfigDirectoryName(), loader.GetConfigFileName());
+                    _hasAttemptedFileWatcherConfiguration = true;
+                    return true;
+                }
+                catch (Exception)
+                {
+                    return false;
```

Comment:
We should at least log a warning here that an attempt to configure hot reloading failed